### DISCUSSION TITLE
Soft Delete for REST API

### DISCUSSION THREAD

**Discussion by lfcc1** [directus/directus#15085] [Archive]
Hello, 
First of all, I would like to congratulate directus team for the excellent work. Been using directus for a month and it's been a pleasant experience.

My question is associated with the soft delete with the REST API calls. I understand that in the data studio I can set a flag "status" that lets me soft delete my collections' entries. However, since I'm using directus as a back end only, I need the soft delete functionality on the REST API. 

When I try to read my collections with the path localhost:8055/items/collection, I get all the items, even the "archived ones" which is not the behavior I would expect/want. (Am I doing something wrong or is this the expected behavior?)

In order to solve this issue I'm planning to:

1. Filter my collections on the role permission (status equals published)
2. Hook filter items.delete to prevent deleting, updating status to "archived" instead.

Is this a good approach?
How could I apply the hook filter to only some collections?

Thank you in advance

**Comment by azrikahar** [comment]
> When I try to read my collections with the path localhost:8055/items/collection, I get all the items, even the "archived ones" which is not the behavior I would expect/want. (Am I doing something wrong or is this the expected behavior?)

It is actually expected! The App does detect whether a collection has archive filter (maybe the `app` in `archive_app_filter` can help to clarify that it is in fact _used in_ the app) then does the filtering behind the scene. In my example below, we can see although we are not viewing archived items by default, the API request actually attaches a filter of `{"status":{"_neq":"archived"}}`. And when we're viewing _all_ including archived items, we can see that there are no filter being used, which is what you're seeing when you say you're getting all items:

https://user-images.githubusercontent.com/42867097/184872623-390076da-ac06-4298-8743-96bdf6e9df1c.mp4

> How could I apply the hook filter to only some collections?

you can either use `<collection>.items.delete` (eg. `articles.items.delete`), or if you're using `items.delete`, you can check the `collection` property in the meta and do something like:

```js
filter('items.delete', (input, { collection }, context) => {
    // example of checking one collection only and doing early return to skip the rest of the logic
    if (collection !== 'recipes') return input;

    // example of checking some collections and doing early return to skip the rest of the logic
    if (['customers', 'articles', 'orders'].includes(collection)) return input;

    /* Additional logic here */

    return input;
});
```

Further reference: https://docs.directus.io/extensions/hooks.html#available-events

**Reply by lfcc1** [reply]
Thank you for the detailed answer.
